src/main/kotlin/no/nav/tag/innsynAareg/service/InnsynService.kt

```diff
@@ -111,6 +118,57 @@ class InnsynService(
         return arbeidsforhold
     }
 
+
+    fun hentTidligereArbeidsforhold
+            (        bedriftsnr: String,
+                    overOrdnetEnhetOrgnr: String,
+                    idPortenToken: String,
+                    fnr:String
+            ): ArbeidsforholdOppslagResultat {
+
+        var arbeidsforhold: ArbeidsforholdOppslagResultat = aaregClient.hentArbeidsforhold(bedriftsnr, overOrdnetEnhetOrgnr, idPortenToken)
+        if (!(arbeidsforhold is ArbeidsforholdFunnet && arbeidsforhold.oversiktOverArbeidsForhold.antall !== null && arbeidsforhold.oversiktOverArbeidsForhold.antall !== 0L)) {
+            arbeidsforhold = finnOpplysningspliktigOgHentArbeidsforhold(bedriftsnr, overOrdnetEnhetOrgnr, idPortenToken, fnr)
+        }
+        if (arbeidsforhold is ArbeidsforholdFunnet) {
+            settNavnPåArbeidsforholdBatch(arbeidsforhold.oversiktOverArbeidsForhold)
+            settYrkeskodebetydningPaAlleArbeidsforhold(arbeidsforhold.oversiktOverArbeidsForhold)
+            return arbeidsforhold;
+        }
+        return arbeidsforhold
+    }
+
+
+    fun finnOpplysningspliktigOgHentArbeidsforhold(
+            bedriftsnr: String,
+            overOrdnetEnhetOrgnr: String,
+            idPortenToken: String,
+            fnr:String
+    ): ArbeidsforholdOppslagResultat {
+        val arbeidsforhold = aaregClient.hentArbeidsforhold(bedriftsnr, overOrdnetEnhetOrgnr, idPortenToken)
+        if (arbeidsforhold is ArbeidsforholdFunnet) {
+            return arbeidsforhold
+        }
+        val organisasjonerMedTilgang = altinnClient.hentOrganisasjonerBasertPaRettigheter(fnr , SERVICEKODE_INNSYN_AAREG,SERVICE_EDITION_INNSYN_AAREG)
+        if( organisasjonerMedTilgang is AltinnOppslagVellykket){
+          val juridiskeEnhetermedTilgang = organisasjonerMedTilgang.organisasjoner.filter { it.Type=="Enterprise"}
+          juridiskeEnhetermedTilgang.forEach {
+              try {
+                  val arbeidsforhold = aaregClient.hentArbeidsforhold(bedriftsnr, it.OrganizationNumber!!, idPortenToken)
+                  if (arbeidsforhold is ArbeidsforholdFunnet && arbeidsforhold.oversiktOverArbeidsForhold.antall !== 0L){
```

Comment:
`!==` er referanse-sjekk. Er to boxed 0 garantert å ha samme referanse i java/kotlin?